### ISSUE TITLE
Question: Is resource to record mapping 1:1?

### ISSUE THREAD

**Issue by dimitvassilev** [cloudcreativity/laravel-json-api#130]
Hello,

I might have noticed a potential issue with the resource to record mapping that one provides in the configuration file. In the underlying neomerx library, the mapping passed to the container (or Encoder::instace()) is modelClass->schemaClass and not modelClass->resourceName, as is done in the config (in reverse). This allows that children classes of modelClass are associated with children classes of schemaClass with different schema configurations within but still with one and the same resource name, i.e. the relationship between resource name and schemas can be 1:n, instead of 1:1. As far as I gathered, schema classes here are deduced through the resource names in the configuration and the directory structure convention, which limits the resource name to schema relationship to 1:1. Am I misenterpreting the implementation logic of either of the libraries? If this restriction is indeed in place, is there some way to deduce within the schema class which of the multiple record classes (as an array of classes is supported in the config) will be serialized, so as to mutate the schema logic accordingly?
One more question: am I right to assume that the record classes in the 'resources' configuration array should be present only once, that is, that the values of the arrray should be unique within the array?

Thanks a lot for any clarifications and recommendations!

Edit: I've just realized - yes, obviously there is a way to recognise what class is being serialized, as it is passed to all (almost all anyway) the schema methods. This means the 1:1 restriction is not a functionality-restricting issue, as multiple schema configurations can exist within the same schema class, being conditional on the model class type.  Routes for the relations to the different model class derivations can also be registered manually with the same resource name as default route parameter. Sorry for having bothered you, I guess it's been a long day and my mind is already not too sharp.

**Comment by lindyhopchris** [comment]
Hi!

It's not documented but you can do this in the config when mapping the resources to records:

```php
'resource-name' => [App\Model1::class, App\Model2::class]
```

Which I think answers your question?

**Comment by dimitvassilev** [comment]
Hi,
You are right, of course, that this solves my problem when combinded with inspecting the model class within the schema, only I did not at first realise that I could differentiate models within the schema. At first glance, multiple models to 1 resource name only seemed to allow that both models be presented with one and the same schema definition. The neomerx library requires associating schemas with models (not resource names with models), thus I could create 2 schemas with one and the same resource name that are responsible for 2 different children classes of a base model, so, thinking along these lines, I was wondering if this were possible now that the configuration is done by associating a resource name to a model. Inspecting what record class is passed to the schema, however, allows presenting it differently even though these different presentations are specified within one and the same schema, I just did not think of that at once. The other thing that was not specifically mentioned in the documetation and made me wonder for a while, was that a model class must be unique in the configuration list, otherwise two resource names will be associated with the class and then the schema which should be used for its representation would become ambiguous.
Anyway, thanks a lot for your reply!
